app/views/locations/upload_summary.html.erb

```diff
@@ -0,0 +1,96 @@
+<h1 class="govuk-heading-l">Upload Summary</h1>
+
+<% if @parent_organisation.errors.present? %>
+  <div class="govuk-error-summary" role="alert" data-module="govuk-error-summary" aria-labelledby="error-summary-title">
+    <h2 id="error-summary-title" class="govuk-error-summary__title">
+      There is a problem
+    </h2>
+    <div class="govuk-error-summary__body">
+      <ul class="govuk-list govuk-error-summary__list">
+        <% @parent_organisation.locations.select(&:new_record?).each do |location| %>
+          <% location_errors = location.errors.where(:address).push(*location.errors.where(:postcode)) %>
```

Comment:
maybe: `location.errors.to_hash(true).slice(:postcode, :address)` could do the trick too?